### PR TITLE
Refactor config

### PR DESCRIPTION
**Pull request by Subsegment**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [ ] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #.

# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

**Comment by Subsegment** [comment]
1. split model/schame
2. refactor Duration to CnosDuration, add CnosByteNumer. To parser config
3. refactor DatabaseSchema, remove Option, use Builder mode
4. move config to DatabaseSchema

File: common/models/src/schema/utils.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        } else {
+            write!(f, "{}", format_duration(self.duration))
+        }
+    }
+}
+
+impl CnosDuration {
+    pub const fn new_with_day(day: u64) -> Self {
+        Self {
+            duration: Duration::from_secs(day * 24 * 60 * 60),
```
Comment (zipper-meng) [inline common/models/src/schema/utils.rs:33]: Prevent duration overflows here or at somewhere upstream.

File: common/models/src/schema/tenant.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+}
+
+#[derive(Debug, Default, Clone, Builder, Serialize, Deserialize)]
+#[builder(setter(into, strip_option), default)]
+pub struct TenantOptions {
+    pub comment: Option<String>,
+    pub limiter_config: Option<TenantLimiterConfig>,
+    drop_after: Option<CnosDuration>,
+    // None means now
```
Comment (zipper-meng) [inline common/models/src/schema/tenant.rs:47]: But it's a boolean.

File: query_server/query/src/sql/parser.rs [hunk 3]
```diff
@@ -655,16 +686,23 @@ impl<'a> ExtParser<'a> {
...
                 "expected database option, but found {}",
                 self.parser.peek_token()
             ));
         }
-        Ok(ExtStatement::AlterDatabase(AlterDatabase {
-            name: database_name,
-            options,
-        }))
+        if config.has_some() {
+            return parser_err!("database config is unmodifiable, only can modify database option：TTL, SHARD, VNODE_DURATION, REPLICA".to_string());
```
Comment (zipper-meng) [inline query_server/query/src/sql/parser.rs:697]: Here has a full-width colon: `database option：TTL,`, while other colons are half-width.

Here has all of **database option**: `TTL, SHARD...`, while none of **database config**. Why not tell user what the database options are?

```
database config (PRECISION, MAX_MEMCACHE_SIZE, MEMCACHE_PARTITIONS, WAL_MAX_FILE_SIZE, WAL_FLUSH_TRIGGER_TOTAL_FILE_SIZE, WAL_SYNC, WAL_SYNC_INTERVAL, WAL_REQ_CHANNEL_CAP, STRICT_WRITE, MAX_CACHE_READERS) is unmodifiable, only database option(TTL, SHARD, VNODE_DURATION, REPLICA) is modifiable
```

Comment (Subsegment) [reply query_server/query/src/sql/parser.rs:697]: there are too many config, and config can't be modified, only tell user which can modify

File: common/models/src/schema/database_schema.rs [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DatabaseConfigBuilder {
+    precision: Option<Precision>,
+    max_memcache_size: Option<u64>,
+    memcache_partitions: Option<u64>,
+    wal_max_file_size: Option<u64>,
+    wal_flush_trigger_total_file_size: Option<u64>,
```
Comment (roseboy-liu) [inline common/models/src/schema/database_schema.rs:259]: remove unused config item

File: meta/src/service/init.rs [hunk 5]
```diff
@@ -1,58 +1,112 @@
...
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct MetaInit {
+    pub cluster_name: String,
+    pub admin_user: String,
+    pub admin_pwd: String,
+    pub system_tenant: String,
+    pub default_database: Vec<(String, DatabaseConfig)>,
+}
+
+impl Default for MetaInit {
```
Comment (roseboy-liu) [inline meta/src/service/init.rs:21]: remove unused code

**Comment by roseboy-liu** [comment]
linked with docs pr.

**Review by roseboy-liu — APPROVED** [review]

**Comment by Subsegment** [comment]
https://github.com/cnosdb/docs/pull/310

**Pull request merged**